include/envoy/grpc/async_client_manager.h

```diff
@@ -22,6 +22,8 @@ class AsyncClientFactory {
 
 using AsyncClientFactoryPtr = std::unique_ptr<AsyncClientFactory>;
 
+enum class AsyncClientFactoryClusterChecks { Skip, ValidateStatic, ValidateStaticDuringBootstrap };
```

Comment:
Where is ValidateStaticDuringBootstrap used in prod code? I don't see it?

Reply:
it's effectively used here: https://github.com/envoyproxy/envoy/blob/main/source/server/server.cc#L620..L621 in `factoryForGrpcApiConfigSource`

This is the place where getThreadLocalCluster doesn't work.

Reply:
OK got it. It really seems like we should be able to do something better here but I'm struggling to think of what that better thing is so I agree we can go with this right now. I will take a look once passing CI.